File: LeetCodeProblemsGo/LeetCode1317ConvertIntegerToTheSumOfTwoNoZeroIntegers/LeetCode1317ConvertIntegerToTheSumOfTwoNoZeroIntegers.go
package LeetCode1317ConvertIntegerToTheSumOfTwoNoZeroIntegers

func hasZero(n int) bool {
	for n > 0 {
		if n%10 == 0 {
			return true
		}
		n /= 10
	}
	return false
}
func getNoZeroIntegers(n int) []int {
	for i, e := 1, n/2; i <= e; i++ {
		if !hasZero(i) && !hasZero(n-i) {
			return []int{i, n - i}
		}
	}
	return []int{}
}

File: LeetCodeProblemsCpp/LeetCodeProblemsCpp/Problems/LeetCode1317ConvertIntegerToTheSumOfTwoNoZeroIntegers.cpp
#include "LeetCode1317ConvertIntegerToTheSumOfTwoNoZeroIntegers.h"

#include <string>
using namespace std;

vector<int> LeetCode1317ConvertIntegerToTheSumOfTwoNoZeroIntegers::getNoZeroIntegers(int n)
{
    vector<int> result{};
    for (int i = 0; i <= n / 2; i++)
    {
        if (to_string(i).find('0') == string::npos && to_string(n - i).find('0') == string::npos)
            return {i, n - i};
    }
    return {};
}
